documentation/tutorial.md

```diff
@@ -160,93 +168,77 @@ selectors in Crawly shell.
+    {"Content-Length", "50607"},
     {"Connection", "keep-alive"},
-    {"Set-Cookie",
-     "__cfduid=d4c96698cfcfdfc9c1ef44ecb162b1cce1587119674; expires=Sun, 17-May-20 10:34:34 GMT; path=/; domain=.homebase.co.uk; HttpOnly; SameSite=Lax; Secure"},
-    {"Cache-Control", "no-cache, no-store"},
-    {"Pragma", "no-cache"},
-    {"Expires", "-1"},
     {"Vary", "Accept-Encoding"},
-    ...,
-    {"cf-request-id", "02294d6d6f0000ffd430ad9200000001"}
-  ], 
+    {"Last-Modified", "Thu, 25 Mar 2021 13:59:05 GMT"},
+    {"Accept-Ranges", "bytes"},
+    {"Strict-Transport-Security", "max-age=15724800; includeSubDomains"}
+  ],
   request: %HTTPoison.Request{
     body: "",
     headers: [],
     method: :get,
     options: [],
     params: %{},
-    url: "https://www.homebase.co.uk/our-range/tools"
+    url: "https://books.toscrape.com/"
   },
-  request_url: "https://www.homebase.co.uk/our-range/tools",
+  request_url: "https://books.toscrape.com/",
   status_code: 200
 }
 ```
 
-Using the shell, you can try selecting elements using Floki with the
-response. Let's say that we want to extract all product categories links from the
-page above:
+Now let's start parsing the document and extract the books in it.
 
 ```
-{:ok, document} = Floki.parse_document(response.body)
-document |> Floki.find("section.wrapper") |> Floki.find("div.article-tiles.article-tiles--wide a") |> Floki.attribute("href")
+response = Crawly.fetch("https://books.toscrape.com/")
 
-["/our-range/tools/power-tools", "/our-range/tools/garage-storage",
- "/our-range/tools/hand-tools", "/our-range/tools/tool-storage",
- "/our-range/tools/ladders", "/our-range/tools/safety-equipment-and-workwear",
- "/our-range/tools/work-benches"]
 ```
 
-The result of running the command above is a list of elements which
-contain href attribute of links selected with
-`a.category-block-heading__title` CSS selector. These URLs will be
-used in order to feed Crawly with requests to follow.
-
-In order to find the proper CSS selectors to use, you might find
-useful opening the target page from the shell in your web browser. You
-can use your browser developer tools to inspect the HTML and come up
-with a selector.
-
-Now let's navigate to one of the Homebase product pages and extract
-data from it.
+All books on the page are enclosed in a `.product_pod` class. We can the select the class and pick the information we want from it.
 
 ```
-response = Crawly.fetch("https://www.homebase.co.uk/bosch-universalimpact-800-corded-hammer-drill_p494894")
-
+items =
+  document
+  |> Floki.find(".product_pod")
+  |> Enum.map(fn x ->
+    %{
+      title: Floki.find(x, "h3 a") |> Floki.attribute("title") |> Floki.text(),
+      price: Floki.find(x, ".product_price .price_color") |> Floki.text()
+    }
+  end)
 ```
 
-Extract the `title` with:
+Here we select all products using the class name, iterate through it and pick the title using `Floki.find(x, "h3 a") |> Floki.attribute("title") |> Floki.text()`
```

Comment:
using the class `.product_pod`, then we iterate over each html fragment found and extract out relevant product information. In this case, we want to extract the title and price.

Reply:
Done